system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -64,7 +64,7 @@ int __timedwait_cp(volatile int *addr, int val,
 
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
-	int is_main_thread = emscripten_is_main_browser_thread();
+	int is_main_thread = emscripten_is_main_runtime_thread();
```

Comment:
To be most accurate, this code should read

```c
int can_futex_wait_on_current_thread = !emscripten_is_main_browser_thread() || current_environment_is_node/*node.js main thread can do Atomics.wait() */;
int is_main_runtime_thread = emscripten_is_main_runtime_thread();

...

          if ((!can_futex_wait_on_current_thread || is_main_runtime_thread) || ...) {

                       ...

			// Assist other threads by executing proxied operations that are effectively singlethreaded.
                        if (is_main_runtime_thread) emscripten_main_thread_process_queued_calls();
...

Reply:
I don't think that should be necessary because the `emscripten_futex_wait` is already safe for all on the main thread and will busy look in a tight loop.    There should be no need to also have this outer busy loop that calls emscripten_futex_wait with smaller values.    

In fact, as a followup I think we can remove the `is_main_thread` part of this condition completely, but I'll leave that as a followup.

Reply:
another way of putting it: both code paths here call emscripten_futex_wait... avoiding calling emscripten_futex_wait is not what this code does.   This code instead relies on being able to call emscripten_futex_wait regardless.

Reply:
Oh, good observation - I did not recall that emscripten_futex_wait was already guarded for main thread to be able to call it. lgtm then.